storage/src/main/java/tech/pegasys/artemis/storage/ChainStorageClient.java

```diff
@@ -236,74 +117,13 @@ public void onNewUnprocessedAttestation(Attestation attestation) {
         ALogger.Color.GREEN);
   }
 
-  // VALIDATOR COORDINATOR METHODS:
-
-  /**
-   * Returns the list of attestations that both have slot number less than or equal to slot, and not
-   * included in any processed block. Removes the attestations from the slot sorted queue in the
-   * process.
-   *
-   * @param state
-   * @param slot
-   * @return
-   */
-  public SSZList<Attestation> getUnprocessedAttestationsUntilSlot(
-      BeaconState state, UnsignedLong slot) {
-    SSZList<Attestation> attestations =
-        new SSZList<>(Attestation.class, Constants.MAX_ATTESTATIONS);
-    int numAttestations = 0;
-    while (unprocessedAttestationsQueue.peek() != null
-        && unprocessedAttestationsQueue.peek().getData().getSlot().compareTo(slot) <= 0
-        && numAttestations < Constants.MAX_ATTESTATIONS) {
-
-      Attestation attestation = unprocessedAttestationsQueue.remove();
-      Bytes32 attestationHashTreeRoot = attestation.getData().hash_tree_root();
-      unprocessedAttestationsMap.remove(attestationHashTreeRoot);
-      // Check if attestation has already been processed in successful block
-      if (ChainStorage.get(attestationHashTreeRoot, this.processedAttestationsBitlistMap)
-          .isPresent()) {
-        Bitlist bitlist =
-            ChainStorage.get(
-                    attestation.getData().hash_tree_root(), this.processedAttestationsBitlistMap)
-                .get();
-        List<Integer> oldAttesters = getAttesterIndicesIntoCommittee(bitlist);
-        List<Integer> newAttesters =
-            getAttesterIndicesIntoCommittee(attestation.getAggregation_bits());
-        if (oldAttesters.containsAll(newAttesters)) {
-          continue;
-        }
-      }
-      attestations.add(attestation);
-      numAttestations++;
-    }
-    return attestations;
-  }
-
-  // Memory Cleaning
-
-  public void cleanAttestationsUntilEpoch(UnsignedLong epoch) {
-    Iterator<Bytes32> it = processedAttestationsBitlistMap.keySet().iterator();
-    while (it.hasNext()) {
-      Bytes32 key = it.next();
-      UnsignedLong currentEpoch = processedAttestationsToEpoch.get(key);
-      if (currentEpoch.compareTo(epoch) < 0) {
-        it.remove();
-        processedAttestationsToEpoch.remove(key);
-      }
-    }
-  }
-
   @Subscribe
-  public void onNewSlot(SlotEvent slotEvent) {
-    if (slotEvent
-            .getSlot()
-            .mod(UnsignedLong.valueOf(Constants.SLOTS_PER_EPOCH))
-            .compareTo(UnsignedLong.ZERO)
-        == 0) {
-      UnsignedLong epoch = BeaconStateUtil.compute_epoch_at_slot(slotEvent.getSlot());
-      if (epoch.compareTo(UnsignedLong.ONE) > 0) {
-        cleanAttestationsUntilEpoch(epoch.minus(UnsignedLong.ONE));
-      }
-    }
+  public void onNewAggregateAndProof(AggregateAndProof attestation) {
```

Comment:
(discussion) Doesn't look like this class is doing anything with `AggregateAndProof`'s, maybe the `BeaconChainController` or some other high-level coordination class would be a better place for this?

Reply:
I only put this here because we have the functions onNewBlock and onNewAttestation here that have the same functionality. If we're going to move this, we should probably move them all to a new class. I'm not sure if that should be in the scope of this PR.